planning/behavior_velocity_planner/src/scene_module/speed_bump/manager.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <scene_module/speed_bump/manager.hpp>
+
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace behavior_velocity_planner
+{
+namespace
+{
+using TrafficSignsWithLaneId = std::vector<std::pair<lanelet::TrafficSignConstPtr, int64_t>>;
+using StopLineWithLaneId = std::pair<lanelet::ConstLineString3d, int64_t>;
+TrafficSignsWithLaneId getTrafficSignRegElemsOnPath(
+  const autoware_auto_planning_msgs::msg::PathWithLaneId & path,
+  const lanelet::LaneletMapPtr lanelet_map)
+{
+  TrafficSignsWithLaneId traffic_signs_reg_elems_with_id;
+  std::set<int64_t> unique_lane_ids;
+  for (const auto & p : path.points) {
+    unique_lane_ids.insert(p.lane_ids.at(0));  // should we iterate ids? keep as it was.
+  }
+
+  for (const auto lane_id : unique_lane_ids) {
+    const auto ll = lanelet_map->laneletLayer.get(lane_id);
+
+    const auto tss = ll.regulatoryElementsAs<const lanelet::TrafficSign>();
+    for (const auto & ts : tss) {
+      traffic_signs_reg_elems_with_id.push_back(std::make_pair(ts, lane_id));
+    }
+  }
+
+  return traffic_signs_reg_elems_with_id;
+}
+
+std::vector<StopLineWithLaneId> getStopLinesWithLaneIdOnPath(
+  const autoware_auto_planning_msgs::msg::PathWithLaneId & path,
+  const lanelet::LaneletMapPtr lanelet_map)
+{
+  std::vector<StopLineWithLaneId> stop_lines_with_lane_id;
+
+  for (const auto & traffic_sign_reg_elem_with_id :
+       getTrafficSignRegElemsOnPath(path, lanelet_map)) {
+    const auto & traffic_sign_reg_elem = traffic_sign_reg_elem_with_id.first;
+    const int64_t lane_id = traffic_sign_reg_elem_with_id.second;
+    // Is speed bump sign?
+    if (traffic_sign_reg_elem->type() != "speed_bump_sign") {
```

Comment:
Is it a type already defined in lanelatt2? Is it a new type?
If it is a new type, it needs to be added to the map specifications of autoware.

Reply:
Do you know where the document? @kenji-miyake @xmfcx @mitsudome-r

Reply:
Wasn't found in https://github.com/fzi-forschungszentrum-informatik/Lanelet2.
It would be better to define it as an extension in [lanelet2_extension](https://github.com/autowarefoundation/autoware.universe/tree/main/map/lanelet2_extension).

Reply:
thank you.
This is what I was looking for.
https://github.com/autowarefoundation/autoware.universe/tree/main/map/lanelet2_extension/docs

Reply:
thank you for the feedback. it's not in lanelet2_extension right now, but it makes sense to add it here. I created this module as in the stop line working logic. I added speed bump sign instead of stop sign. And I'll annotate the stop line as a speed bump. These changes will also be added to lanelet2_extension. I am currently preparing a document to add here.

Reply:
[Q] is stopping really desired behavior for speed bump?
and if so , I think it's better to add "speed_bump" tag condition to stop line module.

Reply:
no, stopping behavior will be removed and modified to decrease speed.

Reply:
@beyzanurkaya 
I see , if so you should refactor from stop ~ to slow_down~.

Reply:
@yukkysaito yes, that would be better.

Reply:
I changed to draft :+1:

Reply:
And please make a util instead of copying as much as possible which you can reuse functions from stop line :pray: